src/utils/Users.js

```diff
@@ -0,0 +1,133 @@
+});
+const getPermissionString = (permission, actions, products) => {
+  let actionStr = 'perform any action';
+  let productStr = 'for all products';
+
+  if (actions.length > 0) {
+    actionStr = niceJoin(actions);
+  }
+
+  if (products.length > 0) {
+    const tmp = niceJoin(products);
+
+    productStr = `for ${tmp}`;
+  }
+
+  return permissionStrings(productStr, actionStr)[permission];
+};
+
+const getRolesString = roles => {
+  const joined = niceJoin(Array.from(roles));
+  let roleStr = 'role';
+
+  if (roles.length > 1) {
+    roleStr = 'roles';
+  }
+
+  return `${joined} ${roleStr}`;
+};
+
+const permissionRestrictionMappings = {
+  admin: {
+    restrict_products: true,
+    restrict_actions: false,
+    supported_actions: [],
+  },
+  emergency_shutoff: {
+    restrict_products: true,
+    restrict_actions: true,
+    supported_actions: ['create', 'modify'],
+  },
+  rule: {
+    restrict_products: true,
+    restrict_actions: true,
+    supported_actions: ['create', 'modify', 'delete'],
+  },
+  release: {
+    restrict_products: true,
+    restrict_actions: true,
+    supported_actions: ['create', 'modify', 'delete'],
+  },
+  release_read_only: {
+    restrict_products: true,
+    restrict_actions: true,
+    supported_actions: ['set', 'unset'],
+  },
+  release_locale: {
+    restrict_products: true,
+    restrict_actions: true,
+    supported_actions: ['modify'],
+  },
+  required_signoff: {
+    restrict_products: true,
+    restrict_actions: true,
+    supported_actions: ['create', 'modify', 'delete'],
+  },
+  permission: {
+    restrict_products: true,
+    restrict_actions: true,
+    supported_actions: ['create', 'modify', 'delete'],
+  },
+  scheduled_change: {
+    restrict_products: true,
+    restrict_actions: true,
+    supported_actions: ['enact'],
+  },
+};
+const allPermissions = [
+  'admin',
+  'emergency_shutoff',
+  'rule',
+  'release',
+  'release_read_only',
+  'release_locale',
+  'required_signoff',
+  'permission',
+  'scheduled_change',
+];
+const supportsProductRestriction = permission => {
+  if (!Object.keys(permissionRestrictionMappings).includes(permission)) {
+    return false;
+  }
+
+  return permissionRestrictionMappings[permission].restrict_products;
+};
+
+const supportsActionRestriction = permission => {
+  if (!Object.keys(permissionRestrictionMappings).includes(permission)) {
+    return false;
+  }
+
+  return permissionRestrictionMappings[permission].restrict_actions;
+};
+
+const getSupportedActions = permission => {
+  if (!Object.keys(permissionRestrictionMappings).includes(permission)) {
+    return [];
+  }
+
+  return permissionRestrictionMappings[permission].supported_actions;
+};
+
+// eslint-disable-next-line import/prefer-default-export
+export {
+  getPermissionString,
+  getRolesString,
+  permissionRestrictionMappings,
+  allPermissions,
+  supportsProductRestriction,
+  supportsActionRestriction,
+  getSupportedActions,
+};
```

Comment:
The filename `Users` gives the impression it's a class or a component which it isn't. Maybe we could rename to something like `userUtils`.